x-pack/elastic-agent/pkg/config/config.go

```diff
@@ -116,6 +116,11 @@ func NewConfigFrom(from interface{}, opts ...interface{}) (*Config, error) {
 	}
 	if len(skippedKeys) > 0 {
 		err = cfg.Merge(skippedKeys, ucfg.ResolveNOOP)
+
+		// we're modifying object
+		for k, v := range skippedKeys {
+			data[k] = v
+		}
```

Comment:
I see, because the code removed it from above they needed to be added back in.

I think maybe making the comment a little clearer and with some reference to the `delete(...)` about would be better.